### PR TITLE
ISPN-7592 Use correct default cache for REST tests

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7592
https://issues.jboss.org/browse/ISPN-7602

### CODE REVIEW

**Comment by tristantarrant** [comment]
Had to make some further changes. Should be good now.

**Comment by tristantarrant** [comment]
The REST rolling upgrade test will fail, but that is because there is a different bug

**Comment by tristantarrant** [comment]
Added a fix for the REST store issue

File: persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java [hunk 1]
```diff
@@ -128,6 +125,7 @@ protected void initChannel(SocketChannel ch) {
       } catch (EncoderException e) {
       }
       this.metadataHelper = Util.getInstance(configuration.metadataHelper(), ctx.getCache().getAdvancedCache().getClassLoader());
+      new HttpResponseHandler(); // Initialize the Netty TypeParameterMatcher
```
Comment (wburns) [inline persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java]: Can you add some more comments around this? This line has a really bad code smell. What happens if this line isn't here for example?

Comment (tristantarrant) [inline persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java]: I have put a log in the description of https://issues.jboss.org/browse/ISPN-7602
I know this looks disgusting, but the REST store really needs an overhaul to use pooling (ISPN-7601) which would trigger proper initialization in the right place.

File: persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java [hunk 2]
```diff
@@ -128,6 +125,12 @@ protected void initChannel(SocketChannel ch) {
       } catch (EncoderException e) {
       }
       this.metadataHelper = Util.getInstance(configuration.metadataHelper(), ctx.getCache().getAdvancedCache().getClassLoader());
+      /*
+       * HACK ALERT. Initialize some internal Netty structures early while we are within the scope of the correct classloader to
```
Comment (tristantarrant) [inline persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java]: @wburns better ?

Comment (wburns) [inline persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java]: Seems fine to me

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request closed without merging**